common/datadogSharedLogic.ts

```diff
@@ -0,0 +1,120 @@
+export function checkForMultipleApiKeys(props: DatadogProps) {
+  let multipleApiKeysMessage;
+  if (props.apiKey !== undefined && props.apiKmsKey !== undefined && props.apiKeySecretArn !== undefined) {
+    multipleApiKeysMessage = "`apiKey`, `apiKmsKey`, and `apiKeySecretArn`";
+  } else if (props.apiKey !== undefined && props.apiKmsKey !== undefined) {
+    multipleApiKeysMessage = "`apiKey` and `apiKmsKey`";
+  } else if (props.apiKey !== undefined && props.apiKeySecretArn !== undefined) {
+    multipleApiKeysMessage = "`apiKey` and `apiKeySecretArn`";
+  } else if (props.apiKmsKey !== undefined && props.apiKeySecretArn !== undefined) {
+    multipleApiKeysMessage = "`apiKmsKey` and `apiKeySecretArn`";
+  }
+
+  if (multipleApiKeysMessage) {
+    throw new Error(`${multipleApiKeysMessage} should not be set at the same time.`);
+  }
+}
+
+/*
+ * Separates logic dedicated to checking for missing properties in DatadogProps,
+ * adding their default values if they exist.
+ *
+ * @returns DatadogStrictProps, identical to the DatadogProps type but with properties
+ * that have default values now set to required.
+ */
+export function handleSettingPropDefaults(props: DatadogProps): DatadogStrictProps {
+  let addLayers = props.addLayers;
+  let enableDatadogTracing = props.enableDatadogTracing;
+  let injectLogContext = props.injectLogContext;
+  const logLevel = props.logLevel;
+  let enableDatadogLogs = props.enableDatadogLogs;
+  let captureLambdaPayload = props.captureLambdaPayload;
+
+  if (addLayers === undefined) {
+    log.debug(`No value provided for addLayers, defaulting to ${DefaultDatadogProps.addLayers}`);
+    addLayers = DefaultDatadogProps.addLayers;
+  }
+  if (enableDatadogTracing === undefined) {
+    log.debug(`No value provided for enableDatadogTracing, defaulting to ${DefaultDatadogProps.enableDatadogTracing}`);
+    enableDatadogTracing = DefaultDatadogProps.enableDatadogTracing;
+  }
+  if (injectLogContext === undefined) {
+    log.debug(`No value provided for injectLogContext, defaulting to ${DefaultDatadogProps.injectLogContext}`);
+    injectLogContext = DefaultDatadogProps.injectLogContext;
+  }
+  if (logLevel === undefined) {
+    log.debug(`No value provided for logLevel`);
+  }
+  if (enableDatadogLogs === undefined) {
+    log.debug(`No value provided for enableDatadogLogs, defaulting to ${DefaultDatadogProps.enableDatadogLogs}`);
+    enableDatadogLogs = DefaultDatadogProps.enableDatadogLogs;
+  }
+  if (captureLambdaPayload === undefined) {
+    log.debug(`No value provided for captureLambdaPayload, default to ${DefaultDatadogProps.captureLambdaPayload}`);
+    captureLambdaPayload = DefaultDatadogProps.captureLambdaPayload;
+  }
+
+  return {
+    addLayers: addLayers,
+    enableDatadogTracing: enableDatadogTracing,
+    injectLogContext: injectLogContext,
+    logLevel: logLevel,
+    enableDatadogLogs: enableDatadogLogs,
+    captureLambdaPayload: captureLambdaPayload,
+  };
+}
```

Comment:
New function created, handles setting setting default values to properties not specified by the user. This logic was previously in the `addLambdaFunctions` function in `datadog.ts`

Returns DatadogStrictProps, identical to the existing DatadogProps interface, but with properties that have default values set as required.